lib/inferno/dsl/fhir_evaluator/data_summary.rb

```diff
@@ -0,0 +1,97 @@
+module FhirEvaluator
+  # DataSummary represents the results of performing data characterization.
+  class DataSummary
+    attr_accessor :root_resource_ids, # All Example (root resource) Ids
+                  :root_bundle_resource_ids, # All Example (root resource) Ids that are Bundle
+                  :domain_resource_ids,   # Domain resource Ids from root and child resources (exclude Bundle)
+                  :resource_profile_map,  # Resources and corresponding profiles
+                  :resource_patient_map,  # Resources and corresponding Patient Ids as subject
+                  :resource_subject_map   # Resources and corresponding subject
+
+    def initialize(data)
+      @root_resource_ids = []
+      @root_bundle_resource_ids = []
+      @domain_resource_ids = []
+      @resource_profile_map = []
+      @resource_patient_map = []
+      @resource_subject_map = []
+
+      validate(data)
+      summarize(data)
+    end
+
+    def validate(data)
+      # Check if duplicate Ids exist for same Resource Type in a data set
+      r_ids = data.map { |r| resources_ids(r) }.flatten
+
+      if r_ids.uniq == r_ids
+        puts 'No duplicate Ids found. Proceed to evaluate..'
```

Comment:
There are many `puts` in the current evaluation tool, what would be the strategy to migrate? `cli/execute/console_outputter.rb`?

Reply:
I think the general strategy would be to have the evaluation broken down into steps, and then produce output based on the results of those steps rather than producing the output as part of those steps.

Reply:
The file is deleted. Archived the comments and will come back if we reuse this file.